feat: optionally disable metadata in bytecode

emilianobonassi opened the pull request:
### What I did

Optionally disable metadata in bytecode

### How I did it

Added the flag `--no-bytecode-metadata` to the cli and pass the parameters in the respective components

### How to verify it

Compile a contract with and without the flag. Bytecode produced will include or not the metadata (e.g. vyper signature) at the end

### Commit message

feat: optionally disable metadata in bytecode

### Description for the changelog

cli: optionally disable metadata in bytecode with `--no-bytecode-metadata`

### Cute Animal Picture

![image](https://user-images.githubusercontent.com/8052911/192379144-5b412624-b7ae-49ef-a501-7ebcd512919b.png)

charles-cooper commented on vyper/ir/compile_ir.py:
```diff
@@ -967,7 +967,7 @@ def adjust_pc_maps(pc_maps, ofst):
 
 
-def assembly_to_evm(assembly, pc_ofst=0, insert_vyper_signature=False):
+def assembly_to_evm(assembly, pc_ofst=0, insert_vyper_signature=False, disable_vyper_signature=False):
```
insert_vyper_signature and disable_vyper_signature both being arguments is a little weird. it should be the callers responsibility to decide what the value here should be

emilianobonassi replied:
correct, i agree, this was a short term resolution to avoid refactoring due to the recursive pattern L1011

charles-cooper reviewed (commented):
looks good, left a comment about one signature, and also i think the option should be named something like `--no-signature`, or, `--no-bytecode-metadata`

emilianobonassi commented:
thanks @charles-cooper 

renamed option to `--no-bytecode-metadata` more future proof and universal. left a comment for the tactical solution implemented. updated description.
